docs/guide/uploading-downloading-binary-content.md

```diff
@@ -0,0 +1,147 @@
+# Uploading & Downloading Binary Content | AngularFire Guide
+
+## Table of Contents
+
+* [Overview](#overview)
+* [API Summary](#api-summary)
+* [Uploading files](#uploading-files)
+* [Displaying images with the `firebase-src` directive](#displaying-images-with-the-firebase-src-directive)
+* [Retrieving files from the template](#retrieving-files-from-the-template)
+
+## Overview
+
+Firebase provides [a hosted binary storage service](https://firebase.google.com/docs/storage/)
+which enables you to store and retrieve user-generated content like images, audio and
+video directly from the Firebase client SDK.
+
+Binary files are stored in a Firebase Storage bucket, not in the Realtime Database.
+The file files in your bucket are stored in a hierarchical structure, just like
+in the Realtime Database.
+
+To use the Firebase Storage binding, first [create a Firebase Storage reference](https://firebase.google.com/docs/storage/web/create-reference).
+Then using this reference, pass it into the `$firebaseStorage` service.
+
+```js
+// define our app and dependencies (remember to include firebase!)
+angular
+  .module("sampleApp", [
+    "firebase"
+  ])
+  .controller("SampleCtrl", SampleCtrl);
+
+// inject $firebaseStorage into our controller
+function SampleCtrl($firebaseStorage) {
+  // create a Firebase Storage Reference for the $firebaseStorage binding
+  var storageRef = firebase.storage().ref('userProfiles/physicsmarie');
+  var storage = $firebaseStorage(storageRef);
+}
+SampleCtrl.$inject = ["$firebaseStorage"];
+```
+
+## API Summary
+
+The Firebase Storage service is created with several special $ methods, all of which are listed in the following table:
+
+| Method  | Description |
+| ------------- | ------------- |
+| [`$put(file, metadata)`](/docs/reference.md#putfile-metadata) |	Uploads file to configured path with optional metadata. Returns an AngularFire wrapped UploadTask. |
+| [`$putString(string, format, metadata)`](/docs/reference.md#putstringstring-format-metadata)	| Uploads a upload a raw, base64, or base64url encoded string with optional metadata. Returns an AngularFire wrapped UploadTask. |
+| [`$getDownloadURL()`](/docs/reference.md#getdownloadurl) |	Returns a promise of the download URL for the file stored at the configured path. |
+| [`$getMetadata()`](/docs/reference.md#getmetadata) | Returns a promise of the metadata of the file stored at the configured path. |
+| [`$updateMetadata(metadata)`](/docs/reference.md#updatemetadatametadata) | Updates the metadata of the file stored at the configured path. Returns a promise containing the complete metadata or en error. |
```

Comment:
Suggestion for the second sentence: "Returns a `Promise` containing the updated metadata." It is a given that it will be rejected with an error in the error case.